src/v/raft/replicate_batcher.cc

```diff
@@ -143,16 +169,14 @@ replicate_batcher::do_cache_with_backpressure(
       std::move(u),
       expected_term,
       consistency_lvl,
-      std::nullopt);
+      timeout);
 
     _item_cache.emplace_back(i);
     co_return i;
 }
 
 ss::future<> replicate_batcher::flush(
   ssx::semaphore_units batcher_units, bool const transfer_flush) {
-    auto holder = _bg.hold();
```

Comment:
this looked a little suspicious but i don't know enough to say if its right or wrong

Reply:
it was intentionally removed as now flush happens in background and it is wrapped with gate